libcontainerd/container_linux.go

```diff
@@ -35,6 +35,14 @@ func (ctr *container) clean() error {
 	}
 	return nil
 }
+func (ctr *container) cleanProcess(id string) {
+	if p, ok := ctr.processes[id]; ok {
```

Comment:
Both of them have a client lock by container ID. I could add a comment to this function that expects that a lock has been taken.